src/coreclr/jit/llvm.cpp

```diff
@@ -1929,6 +1947,7 @@ GenTree* Llvm::createStoreNode(var_types nodeType, GenTree* addr, GenTree* data,
     else
     {
         storeNode = new (_compiler, GT_STOREIND) GenTreeStoreInd(nodeType, addr, data);
+        storeNode->gtFlags |= GTF_IND_TGT_NOT_HEAP;
```

Comment:
The helper is a generic `createStoreNode`, so it seems not right to bake into it the assumptions about the target address' location.

Make a new helper, say `createShadowStackStoreNode` that just adds the `GTF_IND_TGT_NOT_HEAP` (which will be valid for the `STORE_OBJ` too)?

Reply:
Thanks, done